tests/image/test_ssim.py

```diff
@@ -52,16 +60,19 @@ def _sk_ssim(preds, target, data_range, multichannel, kernel_size):
         sk_preds = sk_preds[:, :, :, 0]
         sk_target = sk_target[:, :, :, 0]
 
-    return structural_similarity(
-        sk_target,
-        sk_preds,
-        data_range=data_range,
-        multichannel=multichannel,
-        gaussian_weights=True,
-        win_size=kernel_size,
-        sigma=1.5,
-        use_sample_covariance=False,
-    )
+    results = torch.zeros(sk_preds.shape[0])
+    for i in range(sk_preds.shape[0]):
+        results[i] = structural_similarity(
+            sk_target[i],
+            sk_preds[i],
+            data_range=data_range,
+            multichannel=multichannel,
+            gaussian_weights=True,
+            win_size=kernel_size,
+            sigma=1.5,
+            use_sample_covariance=False,
+        )
+    return results
```

Comment:
```suggestion
    return structural_similarity(
        sk_target,
        sk_preds,
        data_range=data_range,
        multichannel=multichannel,
        gaussian_weights=True,
        win_size=kernel_size,
        sigma=1.5,
        use_sample_covariance=False,
    )
```

Reply:
The problem with the old settings is, they compute the wrong thing, as skimage does not support batching. For example, if a 2D image (BxHxW) is tested with the old setting, skimage interprets it as a 3D image, and does 3D SSIM. The results are thus garbage, this is why atol needed to be set higher (if you average over a high number of random distributed variables, it is no wonder that something similar comes out)

Reply:
Yes, you're right. I've had a deeper look at `skimage` package and they don't consider batch as the first dimension. It seems to me that the following should be working out:

```python
def _sk_ssim(preds, target, data_range, multichannel, kernel_size):
    is_3d = len(preds.shape) == 5

    if is_3d:
        sk_preds = preds.permute(0, 2, 3, 4, 1).squeeze(-1).numpy()
        sk_target = target.permute(0, 2, 3, 4, 1).squeeze(-1).numpy()
    else:
        sk_preds = preds.permute(0, 2, 3, 1).squeeze(-1).numpy()
        sk_target = target.permute(0, 2, 3, 1).squeeze(-1).numpy()

    ssim = []
    for i in range(len(sk_preds)):
        ssim.append(
            structural_similarity(
                sk_target[i],
                sk_preds[i],
                data_range=data_range,
                multichannel=multichannel,
                gaussian_weights=True,
                win_size=kernel_size,
                sigma=1.5,
                use_sample_covariance=False,
            )
        )
    return torch.tensor(ssim).mean()
```

Reply:
Looks better!
There are a few further issues:
1) Our current SSIM implementation reduces over all dimension -including the batch dimension- at the end.
2) The multichannel setting for the skimage test will produce erroneous results.
I'm currently working on these, and will update you with a new version later today

Reply:
Great, thank you a lot for all the endavour! I'll just leave it for now as it is. Please, ping me once you have any update! :]

Reply:
However, if these points are general both for 2D and 3D case, it might be better to open a separate issue and PR for these :]